### PR TITLE
Update LanguageVersion enum doc comments

### PR DESCRIPTION
**Pull request by Youssef1313**
New features doc comments are added relying on:

- https://github.com/dotnet/docs/blob/4518bf80b485566676bcca5ac0250ce53e2c3a98/docs/csharp/whats-new/csharp-7-3.md
- https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8
- https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9

### CODE REVIEW

File: src/Compilers/CSharp/Portable/LanguageVersion.cs [hunk 1]
```diff
@@ -119,19 +119,72 @@ public enum LanguageVersion
...
 
         /// <summary>
         /// C# language version 8.0
         /// </summary>
+        /// <remarks>
+        /// <para>Features:</para>
+        /// <list type="bullet">
+        /// <item><description>Readonly members</description></item>
+        /// <item><description>Default interface methods</description></item>
+        /// <item><description>Pattern matching enhancements (Switch expressions, property patterns, tuple patterns, and positional patterns)</description></item>
```
Comment (cston) [inline src/Compilers/CSharp/Portable/LanguageVersion.cs:148]: Lowercase "S" in "switch".

File: src/Compilers/CSharp/Portable/LanguageVersion.cs [hunk 2]
```diff
@@ -119,19 +119,72 @@ public enum LanguageVersion
...
+        /// <list type="bullet">
+        /// <item><description>Records</description></item>
+        /// <item><description>Init only setters</description></item>
+        /// <item><description>Top-level statements</description></item>
+        /// <item><description>Pattern matching enhancements</description></item>
+        /// <item><description>Native sized integers</description></item>
+        /// <item><description>Function pointers</description></item>
+        /// <item><description>Suppress emitting localsinit flag</description></item>
+        /// <item><description>Target-typed new expressions</description></item>
+        /// <item><description>static anonymous functions</description></item>
```
Comment (cston) [inline src/Compilers/CSharp/Portable/LanguageVersion.cs:178]: Uppercase "S" in "static".

File: src/Compilers/CSharp/Portable/LanguageVersion.cs [hunk 3]
```diff
@@ -119,19 +119,72 @@ public enum LanguageVersion
...
+        /// <para>Features:</para>
+        /// <list type="bullet">
+        /// <item><description>Indexing fixed fields does not require pinning</description></item>
+        /// <item><description>ref local variables may be reassigned</description></item>
+        /// <item><description>stackalloc arrays support initializers</description></item>
+        /// <item><description>More types support the fixed statement</description></item>
+        /// <item><description>Enhanced generic constraints</description></item>
+        /// <item><description>Tuples support == and !=</description></item>
+        /// <item><description>Attach attributes to the backing fields for auto-implemented properties</description></item>
+        /// <item><description>in method overload resolution tiebreaker</description></item>
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Compilers/CSharp/Portable/LanguageVersion.cs:132]:
```suggestion
        /// <item><description>Method overload resolution improvements when arguments differ by 'in'</description></item>
```

File: src/Compilers/CSharp/Portable/LanguageVersion.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Compilers/CSharp/Portable/LanguageVersion.cs:148]:
```suggestion
        /// <item><description>Pattern matching enhancements (switch expressions, property patterns, tuple patterns, and positional patterns)</description></item>
```

**Review by cston — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Comment by jcouv** [comment]
Merged/squashed. Thanks @Youssef1313

**Pull request merged**